### PR TITLE
Implement TreeHash for bitfield

### PR DESCRIPTION
**Pull request by macladson**
In preparation for EIP-7495, bitfield has been moved to https://github.com/sigp/ethereum_ssz.

This PR implements `TreeHash` for both `BitVector` and `BitList` for those types.

Requires https://github.com/sigp/ethereum_ssz/pull/25

### CODE REVIEW

File: tree_hash/Cargo.toml
Comment (macladson) [inline tree_hash/Cargo.toml:16]: Once `ethereum_ssz` is merged, this should be switched to whichever release contains the changes. Same for `ethereum_ssz_derive` below.

**Comment by michaelsproul** [comment]
New release is out for `ethereum_ssz`:

https://github.com/sigp/ethereum_ssz/releases/tag/v0.8.0

**Comment by paulhauner** [comment]
I could use this one for https://github.com/sigp/lighthouse/pull/6609, any chance we can get it merged 🙏

My understanding is that @macladson needs to add a tree-hash test to satisfy teh bot.

**Comment by michaelsproul** [comment]
The bot isn't meant to block the merge due to this config:

https://github.com/sigp/tree_hash/blob/12d828cccc0e1f4869769d02321e708a5c26ded1/.github/workflows/test-suite.yml#L52

We can just ignore it for now

File: tree_hash/src/impls.rs
Comment (michaelsproul) [inline tree_hash/src/impls.rs:210]: Wait, wtf is this?

Comment (michaelsproul) [reply tree_hash/src/impls.rs]: Deleted it

**Review by michaelsproul — APPROVED** [review]

**Pull request merged**
